[T-8080][17.0][ADD] account_invoice_report_shipping_ref

Anxo82 opened the pull request:
@HaraldPanten @Jaimermaccione

Jaimermaccione reviewed (changes requested):
@Anxo82 Tested in a local environment. Functionally, everything works correctly. However, in the report, I noticed that the label appears twice, and I can’t identify the reason for this duplication.

From my point of view, having the label shown at the top should be sufficient.
![image](https://github.com/user-attachments/assets/820d90c9-069b-4660-8979-ca17f15a142c)


Let me know what you think.

Anxo82 commented:
@Jaimermaccione the label appears twice because the task requested it to be included also above the invoice lines, but you are right, it was repeated. I have removed it, could you review? Thanks

manuelregidor commented on account_invoice_report_shipping_ref/README.rst:
```diff
@@ -0,0 +1,75 @@
+This module allows each company to define a custom label for displaying
+the delivery address reference on customer invoice PDFs. The reference
+(ref) from the delivery address contact is shown above the invoice
+lines, using the format: : . If no reference is set, nothing is
```
Could you rewrite the format specification?

manuelregidor commented on account_invoice_report_shipping_ref/README.rst:
```diff
@@ -0,0 +1,75 @@
+the delivery address reference on customer invoice PDFs. The reference
+(ref) from the delivery address contact is shown above the invoice
+lines, using the format: : . If no reference is set, nothing is
+displayed. If no label is defined, the default "Shipping Address Ref."
```
the default "Shipping Address Ref." label
I think that sounds beter

manuelregidor commented on account_invoice_report_shipping_ref/README.rst:
```diff
@@ -0,0 +1,75 @@
+Configuration
+=============
+
+To configure this field, go to Invoicing > Configuration > Settings, in
```
I don't get this part.

Anxo82 replied:
@manuelregidor changes done. Thanks

manuelregidor replied:
This part has not been changed. Could you rewrite it? I don't understand what it means...

ValentinVinagre replied:
is it solved?

Anxo82 replied:
Yes, it's been resolved, sorry for not "resolve conversation"

manuelregidor reviewed (changes requested):
Minor comments.

Anxo82 commented:
@Jaimermaccione @manuelregidor could you review?
Thanks

Anxo82 commented:
@manuelregidor I’ve updated the text in the configure.md file. Is it better like this? Thanks

manuelregidor commented:
@Anxo82 amazing

manuelregidor reviewed (approved):
Technical review. LGTM

Anxo82 commented:
@ValentinVinagre functional and technical review: OK. Can you check and merge, please? 
Thanks

ValentinVinagre commented on account_invoice_report_shipping_ref/models/res_company.py:
```diff
@@ -0,0 +1,10 @@
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    invoice_shipping_ref_label = fields.Char(string="Invoice Shipping Ref. Label")
```
This field should allow translations of the content as it will be printed.

ValentinVinagre commented on account_invoice_report_shipping_ref/models/res_config_settings.py:
```diff
@@ -0,0 +1,14 @@
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    invoice_shipping_ref_label = fields.Char(
```
the same as above.

Anxo82 commented:
@ValentinVinagre changes done! Thanks

ValentinVinagre commented:
@Anxo82 @Jaimermaccione @manuelregidor I'm thinking the field could be HTML in case you want to use a different format. What do you think?

manuelregidor commented:
> @Anxo82 @Jaimermaccione @manuelregidor I'm thinking the field could be HTML in case you want to use a different format. What do you think?

Yes, no problem. However, we have to keep in mind that and HTML field is not that easy to modify by the customer. We should check how it looks like in case no HTML tags are used to define the string.

Anxo82 commented:
@ValentinVinagre @manuelregidor I've changed the field to HTML type and updated both the view and the report. @Jaimermaccione could you do the functional review? Thanks

Anxo82 commented:
@manuelregidor @ValentinVinagre @Jaimermaccione I've found an issue in the view after changing the field to HTML type. Once it’s fixed, I will let you know for the review. Thanks

Anxo82 commented:
@manuelregidor @ValentinVinagre I added a method to avoid a line break in the invoice report. 
@Jaimermaccione could do the functional review? Thanks

ValentinVinagre commented:
/sygelbot merge nobump
